### PR TITLE
Transactions: Commit the query result to get it's generated ID.

### PR DESCRIPTION
**Pull request by ilya-lt**
Commit the query result to retrieve it's generated ID.
This ID is later used to update all the queries with the same query hash.

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] New Query Runner (Data Source)
- [ ] New Alert Destination
- [ ] Other

### CODE REVIEW

File: redash/models/__init__.py [hunk 1]
```diff
@@ -315,6 +315,11 @@ def store_result(cls, org, data_source, query_hash, query, data, run_time, retri
                            retrieved_at=retrieved_at,
                            data=data)
         db.session.add(query_result)
+        db.session.commit()  # Make sure the query result is committed and assigned an ID.
+
+        if query_result.id is None:
+            raise Exception("inserted query result id is null for query_hash:" + query_hash)
```
Comment (daTokenizer) [inline redash/models/__init__.py:321]: Wouldn't it be better if we were to be more specific about the type of Exception that is raised here? 
Also, are we sure it is handled in the calling function? what is the desired behaviour in this case?

**Comment by ilya-lt** [comment]
This is a sanity check I'm not expecting it to be raised(so no point in
doing anything fancy), but it's vital that the code doesn't continue in the
case the if statement is true

**Comment by daTokenizer** [comment]
in this case I don't think an Exception should be raised at all, this returns to the question above: what is the desired behaviour in this case? maybe returning some other value here would better serve this purpose?

**Comment by ilya-lt** [comment]
The desired result is for the parent operation to fail visibly and there
would be a clear message why in the logs

**Comment by daTokenizer** [comment]
ok, os in this case just add to the Exception text the query itself, just to save time when searching backwards for the query hash. 
You convinced me that an exception is appropriate here, even if I don't like that it is the generic `Exception` and not something more specific 🤷‍♀

**Comment by dgilboa** [comment]
[reDash srandom slowness - WIP](https://trello.com/c/KsMR6WCV/1604-redash-srandom-slowness-wip)

**Comment by dgilboa** [comment]
@ilya-lt continuing our discussion from yesterday - can we close this PR ?

**Comment by ilya-lt** [comment]
Generally yes, it does fix the log message to not always state null instead
of ID, but that's all it does, don't know if it's worth merging

**Comment by dgilboa** [comment]
@ilya-lt closing this as not relevant atm.

**Pull request closed without merging**